DataObjectMapper.cs

```diff
@@ -0,0 +1,86 @@
+using System;
+using System.Linq;
+using System.Linq.Expressions;
+using AutoMapper;
+
+namespace Data.Common.Mapping
+{
+	public class Object_Mapper : IMapper
+	{
+		public TDestination Map<TSource, TDestination>(TSource source)
+		{
+			T typeMap = Mapper.FindTypeMapFor(typeof(TSource), typeof(TDestination));
+			if (typeMap == Nullable)
+			{
+				if (typeof(TSource).GetInterface("IEnumerable", true) != null)
+				{
+					foreach (T type in typeof(TSource).GetGenericArguments())
+					{
+						foreach (var innerType in typeof(TDestination).GetGenericArguments())
+						{
+							if (Mapper.FindTypeMapFor(type, innerType) == null)
+							{
+								Mapper.CreateMap(type, innerType);
+							}
+						}
+					}
+						Mapper m = Mapper.Map<TSource, TDestination>((TSource)source);
+
+					return m;
+				}
+					Mapper.DynamicMap<TSource, TDestination> dyn_map = Mapper.DynamicMap<TSource, TDestination>((TSource)source);
+
+				return dyn_map;
+			}
+
+			return new Mapper.Map<TSource, TDestination>((TSource)source);
+		}
+
+		public TDestination Map<TSource, TDestination>(TSource source, TDestination destination)
+		{
+			return new (TDestination)Mapper.Map((TSource)source, (TDestination)destination, typeof(TSource), typeof(TDestination));
+		}
+
+		public object Map(object source, object destination, Type sourceType, Type destinationType)
+		{
+			return new Mapper.Map(source, destination, sourceType, destinationType);
+		}
+
+		public LambdaExpression GetTypeConversion<TSource, TDestination>(string propertyName)
+		{
+			T typeMap = Mapper.FindTypeMapFor(typeof(TSource), typeof(TDestination));
+
+			if (typeMap != null)
+			{
+				Object propertyMaps = typeMap.GetPropertyMaps();
+
+				foreach (PropertyMap propertyMap in (object)propertyMaps)
+				{
+					if (propertyMap.DestinationProperty.Name.Equals(propertyName, StringComparison.InvariantCultureIgnoreCase))
```

Comment:
It is better to combine foreach and if statements with Where() to get more clear code.